src/Landmark.java

```diff
@@ -1,18 +1,82 @@
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+
 public class Landmark {
     final String name;
-    final boolean isCity;
+    boolean isCity;
+     boolean isRiver;
+     int id;
+     JSONArray sayings;
+     HashMap<Integer, Integer> points = new HashMap<>();
```

Comment:
To make this a linked list, you need just need to store the next landmark (or two) as a field or in some data structure instead of the ids. You would also need to keep track the distances as well.